native/components/message-markdown.react.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import { Component } from 'react';
+import * as React from 'react';
+import { View, ViewPropTypes } from 'react-native';
+import { merge, pick } from 'lodash';
+import PropTypes from 'prop-types';
+import SimpleMarkdown from 'simple-markdown';
+
+import styles from '../themes/markdown-styles';
+import darkStyles from '../themes/markdown-dark-styles';
+import rules from './message-markdown-rules.react';
+
+type Props = {
+  onLink: (url: string) => Promise<any>,
+  styles: typeof styles,
+  rules: any,
+  darkStyle: boolean,
+  children: React.Node,
+};
+class Markdown extends Component<Props> {
+  static propTypes = {
+    onLink: PropTypes.func,
+    styles: PropTypes.shape({
+      view: ViewPropTypes.style,
+    }),
+    rules: PropTypes.object,
+    darkStyle: PropTypes.bool,
+    children: PropTypes.node,
+  };
+
+  static defaultProps = {
+    styles: styles,
+    rules: {},
+    darkStyle: false,
+  };
+  parse: any;
+  renderer: any;
+
+  constructor(props: Props) {
+    super(props);
+    const opts = {
+      onLink: props.onLink,
+      rules: props.rules,
+    };
+
+    const style = props.darkStyle ? darkStyles : styles;
+    var customRules = rules(style, opts);
+
+    var customMarkdownRules = pick(SimpleMarkdown.defaultRules, [
+      'autolink',
+      'link',
+      'reflink',
+      'text',
+      'paragraph',
+      'url',
+    ]);
+
+    customRules = merge({}, customMarkdownRules, customRules, opts.rules);
```

Comment:
It looks like rules can come from way too many places. Let's simplify this logic. Can we keep all of our rules in `message-markdown-rules.react.js`?

Reply:
I guess we can get rid of `opts.rules`, since we don't need to add more based on component we're creating